include/deal.II/base/quadrature.h

```diff
@@ -180,7 +180,7 @@ class Quadrature : public Subscriptor
    * to this object.
    */
   Quadrature &
-  operator=(Quadrature<dim> &&) = default; // NOLINT
+  operator=(Quadrature<dim> &&) noexcept = default;
```

Comment:
The standard says that `noexcept` is inferred automatically. `clang-tidy` still wants us to be explicit here. If we are guessing wrong, the compiler would complain just as it does for the move constructor of `FiniteElement`. Hence, both of the above versions are fine. The intent of this PR was to use `NOLINT` at as few places as possible so I decided for the `noexcept` version.

Reply:
Right. So the standard allows adding `noexcept` iff it could have inferred that by itself, and doing so is then really just a way to make it clearer without actually changing any semantics?

What did `clang-tidy` complain about before? That we *should* have marked the operator as `noexcept` but didn't?

Reply:
Yes, exactly.

Reply:
Hm, how unnecessary. But ok -- I'm all for being explicit, and a throwing move constructor is probably not very useful anyway :-)